crypto/src/signature.rs

```diff
@@ -42,12 +42,36 @@ pub struct PublicKey3(pub CompressedCurvePoint);
 #[wire_type(recurse = 2)]
 pub struct Signature((FieldPoint, InnerCurveScalar));
 
+impl Signature {
+    /// field_point
+    pub fn field_point(&self) -> &FieldPoint {
+        &self.0 .0
+    }
+
+    /// inner_curve_scalar
+    pub fn inner_curve_scalar(&self) -> &InnerCurveScalar {
+        &self.0 .1
+    }
+}
+
 #[derive(Clone, Serialize, Deserialize, Default, PartialEq, Debug)]
 pub struct FieldPoint(BaseHash);
```

Comment:
This will need to be unified with the changes in #81 once that is merged I think https://github.com/ChainSafe/mina-rs/blob/685a9b649ad6cbc1050337132135c776b3365b8d/base/src/protocol_state_proof/field_and_curve_elements.rs#L36

We should probably refactor the EC primitives into a different crate

Reply:
Sure, thx for reminding